src/System.Private.ServiceModel/tests/Common/Infrastructure/ServiceUtilHelper.cs

```diff
@@ -280,49 +274,17 @@ public static void EnsurePeerCertificateInstalled()
     // propagated back to the caller.
     private static X509Certificate2 InstallClientCertificateFromServer()
     {
-        X509Certificate2 clientCertificate = null;
-        BasicHttpBinding basicHttpBinding = new BasicHttpBinding();
-        ChannelFactory<IUtil> factory = null;
-        IUtil serviceProxy = null;
-        try
-        {
-            factory = new ChannelFactory<IUtil>(basicHttpBinding, new EndpointAddress(ServiceUtil_Address));
-            serviceProxy = factory.CreateChannel();
-            byte[] certdata = serviceProxy.GetClientCert(false);
-            clientCertificate = new X509Certificate2(certdata, "test", X509KeyStorageFlags.PersistKeySet);
-            clientCertificate = CertificateManager.InstallCertificateToMyStore(clientCertificate);
-        }
-        finally
-        {
-            CloseCommunicationObjects((ICommunicationObject)serviceProxy, factory);
-        }
-
-        return clientCertificate;
+        X509Certificate2 clientCertificate = new X509Certificate2(GetResourceFromServiceAsByteArray(ClientCertificateResource), "test", X509KeyStorageFlags.PersistKeySet);
+        return CertificateManager.InstallCertificateToMyStore(clientCertificate);
     }
 
     // Acquires the peer trust certificate from the service utility endpoint and
     // attempts to install it into the TrustedPeople store.  All failures are
     // propagated back to the caller.
     private static X509Certificate2 InstallPeerCertificateFromServer()
     {
-        X509Certificate2 peerCertificate = null;
-        BasicHttpBinding basicHttpBinding = new BasicHttpBinding();
-        ChannelFactory<IUtil> factory = null;
-        IUtil serviceProxy = null;
-        try
-        {
-            factory = new ChannelFactory<IUtil>(basicHttpBinding, new EndpointAddress(ServiceUtil_Address));
-            serviceProxy = factory.CreateChannel();
-            byte[] certdata = serviceProxy.GetPeerCert(false);
-            peerCertificate = new X509Certificate2(certdata, "test", X509KeyStorageFlags.PersistKeySet);
-            peerCertificate = CertificateManager.InstallCertificateToTrustedPeopleStore(peerCertificate);
-        }
-        finally
-        {
-            CloseCommunicationObjects((ICommunicationObject)serviceProxy, factory);
-        }
-
-        return peerCertificate;
+        X509Certificate2 peerCertificate = new X509Certificate2(GetResourceFromServiceAsByteArray(PeerCertificateResource), "test", X509KeyStorageFlags.PersistKeySet);
```

Comment:
When I get the dependencies working better, we fail at this line:

WindowsCryptographicException
Cannot find the requested object
   at Internal.Cryptography.Pal.CertificatePal.FromBlobOrFile(Byte[] rawData, String fileName, String password, X509KeyStorageFlags keyStorageFlags)
   at System.Security.Cryptography.X509Certificates.X509Certificate..ctor(Byte[] rawData, String password, X509KeyStorageFlags keyStorageFlags)
   at ServiceUtilHelper.InstallPeerCertificateFromServer() in c:\OSS\wcf\src\System.Private.ServiceModel\tests\Common\Infrastructure\ServiceUtilHelper.cs:line 286
   at ServiceUtilHelper.EnsurePeerCertificateInstalled() in c:\OSS\wcf\src\System.Private.ServiceModel\tests\Common\Infrastructure\ServiceUtilHelper.cs:line 236